cache/cache.go

```diff
@@ -0,0 +1,186 @@
+package cache
+
+import (
+	"bytes"
+	"sort"
+	"sync"
+
+	"go.uber.org/atomic"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+type ATXData struct {
+	Weight             uint64
+	BaseHeight, Height uint64
+	Nonce              types.VRFPostIndex
+	Malicious          bool
```

Comment:
i think nonce and malicious is a property of identity, not atx.

if an identity is marked malicious in epoch 3, it should still be malicious in epoch 6. but with current design if nodes didn't restart btwn these time it will be forgotten.

Reply:
it doesn't matter whose property logically is it. it is about how data is stored and indexed.
identity is relevant in consensus only if it has atx in this epoch.

Reply:
> identity is relevant in consensus only if it has atx in this epoch.

this is theoretically true, but in practice we already have this in hare/tortoise where we use nodeid->malicious mapping to avoid checking nodeid->"atx in this epoch" lookup.

and cachedb's IsMalicious() is also by nodeid. how do you plan to replace that?

Reply:
hare GetByNode, tortoise Get

Reply:
specifically hare will use only fastpath that relies on consensus data, it won't even need to have references to db for that.

tortoise (proposal/ballots handler) will query using Get(atx)